app/controllers/posts_controller.rb

```diff
@@ -64,18 +71,27 @@ def create
 def update
 	set_post
 	if @post.update(post_params)
-    redirect_to posts_path, notice: "Successfully updated post."
-  else
-  	@all_needs = Need.by_category
-		@post_need = @post.post_needs.build
-    render action: "edit"
-  end
+	    redirect_to posts_path, notice: "Successfully updated post."
+	else
+	  	@all_needs = Need.by_category
+			@post_need = @post.post_needs.build
+	    render action: "edit"
+	end
 end
 
 def update_needs
 	post_id = params['post']['id'].to_i
 	@post = Post.find(post_id)
 	@post.update_post_needs(params['post_needs']['completed_ids'])
+	@post.date_completed = DateTime.current
+	@post.post_needs.each do |n|
```

Comment:
Let's move this check into the Post model so that the controller code reads like: @post.date_completed = DateTime.current if @post.all_needs_completed?